uitools/cpp/Esri/ArcGISRuntime/Toolkit/BasemapGalleryController.cpp

```diff
@@ -607,15 +641,46 @@ namespace Toolkit {
       if (item && !it_sp.isEmpty())
         return sp == item->spatialReference();
     }
+
+    //scene case:
+    const auto layers = basemap->baseLayers();
+    if(layers->size() <= 0)
+      return false;
+
+    if(auto sceneView = static_cast<Scene*>(m_geoModel))
```

Comment:
💭 Should it not actually build if it is not able to cast to the specified type when using the static_cast? How can get a segmentation fault?

Reply:
We have

```
class Map: public GeoModel
```

and

```
class Scene: public GeoModel
```

The given GeoModel can be either a Map or a Scene. Here we are casting the GeoModel to a Scene without any test to see if it might be a Map instead. `static_cast` is a compile-time check only, not a runtime check, and casting a GeoModel to a Scene is always a valid type conversion regardless of the underlying data.

Reply:
This might help illustrate https://onlinegdb.com/mQheT-BDB